### ISSUE TITLE
Hi, a question about clustering assignments C?

### ISSUE THREAD

**Issue by A11en0** [penghu-cs/MRL#7]
Thanks for your wonderful job and great paper writing.

But I'm confused about why does the performance promote? Because after I read the codes, I realize that the clustering assignment C is just a linear layer like another classification model. It's no different from the normal way. On the other hand, RC loss may be more suitable for noise settings than CE loss, However, its working role is also the same as the CE loss. I guess it's not the key to such a great performance. So, my question is, how does the model deal with noise so efficiently.

Hope for your reply, thanks again.

**Comment by penghu-cs** [comment]
Thanks. The normalized clustering assignment C is the center of each class, thus we could use cosine to measure the similarity across different samples. Moreover, C is updated by descending gradients. Because of the memory effect of neural networks, the model will capture the correct information from noisy data in the first epochs, however, CE loss will overemphasize the noise in the further training. To alleviate the overemphasizing problem of CE loss, RC loss is present to alleviate the focus on hard samples that probably are noisy data. Therefore, our method could be more robust against noisy labels.

**Comment by A11en0** [comment]
Thanks for your clear explanation! I am trying to use the clustering assignment trick in my own model, but it's a multi-label task, I am a little bit confusing whether should I modify Eq.3 in the paper?

**Comment by penghu-cs** [comment]
I think you could modify it for multi-label cases since it is designed for single-label cases.
